libs/libarchfpga/src/physical_types.h

```diff
@@ -653,6 +653,9 @@ struct t_physical_tile_type {
 
     // Does this t_physical_tile_type contain an outpad?
     bool is_output_type = false;
+
+    // Is this t_physical_tile_type an empty type?
+    bool is_empty = false;
```

Comment:
When we will refactor the data structure, I think I will not keep a boolean variable. Currently, as far as I see, to identify an empty ``physical_tile``, there are multiple ways. I see in some codes, the name is used as a reference, while in other codes, the index is used as a reference. That will create issues. That is why we need an API ``is_empty()`` to provide a general reference. I do not have a strong opinion whether to use name or index. But we should use an API so that developers have a standard way to follow.

For example:

```
bool is_empty() {
  return std::string(name)== EMPTY_BLOCK_NAME;
} 
```

I am open to discussion. Let me know what you think.

Reply:
Right, I have actually implemented this change in https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1937/commits/162ca071680bcd720af4421b29cffc5688644383. Also, I think that for now, the Interchange can also make use of the default name for the `EMPTY` blocks, but it is worth noticing that, if this can be a customizable parameter, relying on its name to verify whether a block is empty could not be the best solution.

I am thinking that maybe we can use the same approach as for the [architecture name](https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/36789e456ab292038b8518bf2c835c9b9dcaaa0e/libs/libarchfpga/src/arch_util.cpp#L23-L36)?

Reply:
@acomodi The approach for the architecture name makes sense to me. We follow this way.